### ISSUE TITLE
Docstring for `Canvas.transform` Should include a blurb on idiomatically creating matrices

### ISSUE THREAD

**Issue by 4amVim** [flutter/flutter#83179]
## Use case
[`Canvas.transform()`](https://api.flutter.dev/flutter/dart-ui/Canvas/transform.html) accepts a 4d matrix, but doesn't indicate any way to do it. Perusing only on documentation, only route forward is with `Float64List`. 

Creating matrices using [`Float64List`](https://api.flutter.dev/flutter/dart-typed_data/Float64List-class.html) is tedious, and it hampers code-readability.
Yet, its the only way somebody just reading docs would construct a matrix for `transform`.

## Proposal
Given the extreme utility of `transform`, and its indispensability, in lots of use-cases, we should indicate a convenient way to use it.
Currently, the (vector_math)[https://pub.dev/packages/vector_math] package offers a great way to create [`Matrix4`](https://pub.dev/documentation/vector_math/latest/vector_math/Matrix4-class.html)
 in a variety of ways, which can then be passed to `transform` via `.storage`. Moreover the package is maintained by Google.

I'd be more than happy to submit a PR for the same.

**TL;DR: Should add a mention of vector_math to the docstring of `Canvas.transform`, and add a simple example as well.**

**Comment by darshankawar** [comment]
@4amVim 
Does [this](https://github.com/flutter/flutter/issues/38721) issue sound similar to your request ?

**Comment by 4amVim** [comment]
They both are about Canvas' internal matrix yes. 
But this issue proposes that we should include a bit on how to use vector_math to construct a matrix. Purely documentation, so that a developer can stumble upon good practices easier.

While that issue proposes that the internal matrix be exposed. Which is additional functionality.

**Comment by shtse8** [comment]
Any documentation about the transform matrix?
